src/Repo_new.java

```diff
@@ -0,0 +1,804 @@
+import java.io.File;
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+import com.starbase.starteam.CheckinManager;
+import com.starbase.starteam.CheckinOptions;
+import com.starbase.starteam.CheckoutManager;
+import com.starbase.starteam.CheckoutOptions;
+import com.starbase.starteam.Folder;
+import com.starbase.starteam.Item;
+import com.starbase.starteam.LogonException;
+import com.starbase.starteam.Project;
+import com.starbase.starteam.Server;
+import com.starbase.starteam.ServerInfo;
+import com.starbase.starteam.StarTeamFinder;
+import com.starbase.starteam.StarTeamURL;
+import com.starbase.starteam.User;
+import com.starbase.starteam.View;
+import com.starbase.starteam.ViewConfiguration;
+import com.starbase.starteam.vts.comm.NetMonitor;
+import com.starbase.util.OLEDate;
+
+/**
+ * @author Andrey Pavlenko
+ */
+public class Repo implements AutoCloseable {
+    private static final String FOOTER_START = "----------------------------------- StarTeam -----------------------------------";
+    private static final String FOOTER_END = "--------------------------------------------------------------------------------";
+    public static final String LS = System.getProperty("line.separator");
+    public static final String DATE_FORMAT = "dd.MM.yy HH:mm:ss";
+    private final RepoProperties _repoProperties;
+    private final Logger _logger;
+    private final ConnectionPool _pool;
+    private final List<View> _viewCache;
+    private final String _branchName;
+    private final String _userNamePattern;
+    private final Map<String, Folder> _folderCache;
+    private final Map<String, com.starbase.starteam.File> _fileCache;
+    private final MessageFormat _commentFormat;
+    private volatile List<Pattern> _ignoreFiles;
+    private View _view;
+    private Folder _rootFolder;
+    private File _tempDir;
+
+    public Repo(final RepoProperties repoProperties, final Logger logger) {
+        _branchName = repoProperties.getBranchName();
+        _userNamePattern = repoProperties.getProperty(PROP_USER_PATTERN,
+                PROP_DEFAULT_USER_PATTERN);
+        _repoProperties = repoProperties;
+        _logger = logger;
+        _pool = new ConnectionPool();
+        _viewCache = new ArrayList<>();
+        _folderCache = new ConcurrentHashMap<>();
+        _fileCache = new ConcurrentHashMap<>();
+        //@formatter:off
+        _commentFormat = new MessageFormat(
+           "{0}\n\n"
+         + FOOTER_START + '\n'
+         + "{1}\n"
+         + FOOTER_END);
+      //@formatter:on
+    }
+
+    public Server getIdleConnection() {
+        return _pool.get();
+    }
+
+    public void releaseConnection(final Server s) {
+        _pool.release(s);
+    }
+
+    private Server createNewConnection() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move this method into "ConnectionPool". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS3398)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "createNewConnection" is 24 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)